app/controllers/posts_controller.rb

```diff
@@ -8,4 +8,26 @@ def show
     @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
   end
+
+  def new
+    @user = current_user
```

Comment:
- [ ] [optional] @muneeb3123 Great work on your posts_controller however you do not have to create @user in your new method as you already did it in your create method making its creation redundant.